src/main/java/com/arrl/radiocraft/common/radio/voice/handheld/PlayerRadio.java

```diff
@@ -162,63 +183,67 @@ public boolean openChannel() {
     }
 
     @Override
-    public void receive(AntennaVoicePacket antennaPacket) {
-        if(isReceiving()) {
-            Player player = getPlayer();
-            if(player == null){
-                Radiocraft.LOGGER.error("receiving but player is null?");
-                return;
-            }
-            //TODO remove
-            System.err.println("Receiving audio length " + antennaPacket.getRawAudio().length + " strength " + antennaPacket.getStrength() + " player " + player.getName() + " " + player.getUUID() + " eye position of player" + player.getEyePosition() + " from " + antennaPacket.getSourcePlayer());
-            if(receiveChannel == null)
-                if(!openChannel()) return; //if the channel cannot be opened, return early
+    public synchronized void receive(AntennaVoicePacket antennaPacket) {
+        if(this.isReceiving()) {
+//            System.err.println("Receiving audio length " + antennaPacket.getRawAudio().length + " strength " + antennaPacket.getStrength() + " player " + player.getName() + " " + player.getUUID() + " eye position of player" + player.getEyePosition() + " from " + antennaPacket.getSourcePlayer());
 
             //if entityChannel can be gotten working again, this is where you check to make sure it's still bound to the player
-            //most likely redundant with the onPlayerCloned hook in PlayerRadioManager calling set player
 //            if(receiveChannel.getEntity().getEntity() != player){
 //                System.err.println("receivechannel entity didn't equal this player? entity: " + receiveChannel.getEntity().getEntity() + " player " + player);
 //                receiveChannel.updateEntity(RadiocraftVoicePlugin.API.fromEntity(player));
 //            }
-            if(currentlevel != player.level()){
-                receiveChannel.flush();
+
+
+            if (voiceChannelCurrentLevel != this.voiceLevel) {
+                if (receiveChannel != null) receiveChannel.flush();
                 receiveChannel = null;
-                if(!openChannel()) return;
+                if (!openChannel()) return;
+            }else if(receiveChannel == null) {
+                if (!openChannel()) return;
+            } else {
+                receiveChannel.updateLocation(getPosInVoiceApiFormat());
             }
-            receiveChannel.updateLocation(getPosInVoiceApiFormat());
 
+            long runningTotal = 0;
             short[] rawAudio = antennaPacket.getRawAudio();
-            for(int i = 0; i < rawAudio.length; i++)
-                rawAudio[i] = (short)Math.round(rawAudio[i] * antennaPacket.getStrength()); // Apply appropriate gain for signal strength
+            for(int i = 0; i < rawAudio.length; i++) {
+                short sample = (short) Math.round(rawAudio[i] * antennaPacket.getStrength()); // Apply appropriate gain for signal strength
+                runningTotal = sample * sample;
+                rawAudio[i] = sample;
+            }
```

Comment:
This is one of those scenarios where it *might* be faster to store the value of `antennaPacket.getStrength()` to a variable defined before the for loop, to avoid making 960 calls to `getStrength()` during the processing of every single 20ms voice packet.